geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -110,6 +111,42 @@ private static MemberSet calculateDiff(RegionProvider regionProvider, List<Redis
     return diff;
   }
 
+  public static Set<byte[]> sinter(RegionProvider regionProvider, List<RedisKey> keys) {
+    List<RedisSet> sets = new ArrayList<>(keys.size());
+    RedisSet smallestSet = NULL_REDIS_SET;
```

Comment:
I wonder if this code would be easier to understand if you just did the following:
```
List<RedisSet> sets = createRedisSetList(keys, regionProvider);
RedisSet smallestSet = findSmallest(sets);
if (smallestSet.scard() == 0) {
  return emptySet();
}
```
now create the result with the existing code except as you iterate otherSet checks to see if "otherSet == smallestSet" and if so skip it.